modules/web/js/ballerina/views/ballerina-file-editor.js

```diff
@@ -1070,27 +1065,81 @@ class BallerinaFileEditor extends BallerinaView {
         this._sourceView.clearExistingDebugHit();
     }
 
-    _showBreakpoint(newBreakpoint) {
-        var modelMap = this.diagramRenderingContext.getViewModelMap();
-        var self = this;
-        this._currentBreakpoints = this._currentBreakpoints || [];
-
-        _.each(modelMap, function(aView) {
-            if(!_.isNil(aView.getModel)) {
-                var lineNumber = aView.getModel().getLineNumber();
-                if(newBreakpoint === lineNumber && !_.isNil(aView.showDebugIndicator)) {
-                    aView.showDebugIndicator();
-                    self._currentBreakpoints.push(aView);
-                }
+     /**
+      * highlights breakpoints in designview
+     */
+    _showDesignViewBreakpoints(breakpoints = []) {
+        const findBreakpointsVisitor = new FindBreakpointsVisitor(this._model);
+        findBreakpointsVisitor.setBreakpoints(breakpoints);
+
+        var generatedSource = this.getContent();
+        var model = this.getModelFromSource(generatedSource);
+        this.setModel(model);
+        this.reDraw();
+        const modelMap = this.diagramRenderingContext.getViewModelMap();
+        model.accept(findBreakpointsVisitor);
+        const breakpointNodes = findBreakpointsVisitor.getBreakpointNodes();
+        // const fileName = this._file.getName();
+
+        // hide previous breakpoints first
+        this.hideCurrentBreakpoints();
+
+        breakpointNodes.forEach( breakpointNode => {
+            const nodeView = modelMap[breakpointNode.id];
+            if(!nodeView) {
+                return;
             }
+            nodeView.showDebugIndicator = nodeView.showDebugIndicator || function() {};
+            nodeView.showDebugIndicator();
+            // DebugManager.addBreakPoint(breakpointNode.getLineNumber(), fileName);
+            const pathVector = [];
+            this.getPathToNode(breakpointNode, pathVector);
+            this._currentBreakpoints.push(JSON.stringify(pathVector));
         });
     }
 
-    _hideBreakpoint() {
+    _showSourceViewBreakPoints() {
+        const pathVectors = this.getCurrentBreakpoints() || [];
+        const breakpoints = [];
+        pathVectors.forEach(pathVectorStr => {
+            const pathVector = JSON.parse(pathVectorStr);
+            const node = this.getNodeByVector(this._model, pathVector);
+            const lineNumber = node.getLineNumber();
+            breakpoints.push(lineNumber);
+        });
+        this._sourceView.setBreakpoints(breakpoints);
+    }
+
+    hideCurrentBreakpoints() {
```

Comment:
fixed in https://github.com/ballerinalang/composer/commit/816f10086d7ab68eaa48db0862e8a9f48986c8c1